### PR TITLE
Projects redesign/prevent sudden zoom

### PR DESCRIPTION
**Pull request by sunilsabatp**
- [x] -  Zoom to center of results on search/filter??
- [x] -  Check zoom behavior for zooming into site, should stop before last "jump"

### CODE REVIEW

**Comment by mohitb35** [comment]
Functionality works perfectly, good work @sunilsabatp

**Review by mariahosfeld — COMMENTED** [review]
Functionality looks good to me

File: src/features/projectsV2/ProjectsMap/index.tsx [hunk 1]
```diff
@@ -1,18 +1,24 @@
 import 'maplibre-gl/dist/maplibre-gl.css';
 import Map, { NavigationControl } from 'react-map-gl-v7/maplibre';
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useRef, MutableRefObject } from 'react';
 import { useProjectsMap } from '../ProjectsMapContext';
```
Comment (mohitb35) [inline src/features/projectsV2/ProjectsMap/index.tsx:5]: Clean up unwanted deps

File: src/utils/projectV2.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 } from '../features/common/types/projectv2';
 import {
   PlantLocation,
   PlantLocationSingle,
   SamplePlantLocation,
 } from '../features/common/types/plantLocation';
-import { NextRouter } from 'next/router';
+import * as turf from '@turf/turf';
+import { Position } from 'geojson';
+import { MapRef } from './mapsV2/zoomToProjectSite';
```
Comment (mohitb35) [inline src/utils/projectV2.ts:17]: MapRef needs to be imported from /types/projectv2

File: src/utils/projectV2.ts [hunk 3]
```diff
@@ -142,3 +144,54 @@ export const getPlantData = (
...
+ */
+export const centerMapOnCoordinates = (
+  mapRef: MapRef,
+  [longitude, latitude]: Position
+) => {
+  if (!mapRef.current) return;
+  mapRef.current.getMap().easeTo({
+    center: [longitude, latitude],
+    duration: 1200,
+    easing: (t) => t * (2 - t),
```
Comment (mohitb35) [inline src/utils/projectV2.ts:195]: Resolve the TS warning here

**Review by mohitb35 — APPROVED** [review]
@sunilsabatp Minor TS warnings to be addressed. LGTM after that's done, no need for another review.

**Pull request merged**